crates/ruff/src/rules/flake8_todos/rules.rs

```diff
@@ -223,27 +223,70 @@ impl Violation for MissingSpaceAfterTodoColon {
     }
 }
 
-static TODO_REGEX_SET: Lazy<RegexSet> = Lazy::new(|| {
-    RegexSet::new([
-        r#"^#\s*(?i)(TODO).*$"#,
-        r#"^#\s*(?i)(FIXME).*$"#,
-        r#"^#\s*(?i)(XXX).*$"#,
-    ])
-    .unwrap()
-});
+enum Directive {
+    Todo,
+    Fixme,
+    Xxx,
+}
 
-// Maps the index of a particular Regex (specified by its index in the above TODO_REGEX_SET slice)
-// to the length of the tag that we're trying to capture.
-static PATTERN_TAG_LENGTH: &[usize; 3] = &["TODO".len(), "FIXME".len(), "XXX".len()];
+impl Directive {
+    /// Extract a [`Directive`] from a comment.
+    ///
+    /// Returns the offset of the directive within the comment, and the matching directive tag.
+    fn from_comment(comment: &str) -> Option<(usize, Directive)> {
```

Comment:
Use `TextSize` for offsets to make it clear, that this is a text offset (and e.g. not an offset into an array)

Reply:
https://github.com/charliermarsh/ruff/pull/4426